grafana/resource_library_panel.go

```diff
@@ -0,0 +1,260 @@
+package grafana
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	gapi "github.com/grafana/grafana-api-golang-client"
+)
+
+func ResourceLibraryPanel() *schema.Resource {
+	return &schema.Resource{
+
+		Description: `
+Manages Grafana library panels.
+
+* [Official documentation](https://grafana.com/docs/grafana/latest/panels/panel-library/)
+* [HTTP API](https://grafana.com/docs/grafana/latest/http_api/library_element/)
+`,
+
+		CreateContext: CreateLibraryPanel,
+		ReadContext:   ReadLibraryPanel,
+		UpdateContext: UpdateLibraryPanel,
+		DeleteContext: DeleteLibraryPanel,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"uid": {
+				Type:     schema.TypeString,
+				Computed: true,
```

Comment:
I believe you need to set `Optional: true` as well if you want the UID to be settable